app/code/Magento/NewsletterApi/Api/SubscriptionManagementInterface.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\NewsletterApi\Api;
+
+use Magento\Customer\Api\Data\CustomerInterface;
+
+/**
+ * Newsletter Subscription Management Interface
+ *
+ * @api
+ */
+interface SubscriptionManagementInterface
+{
+    /**
+     * Subscribe
+     *
+     * Adds a new Newsletter Subscription
+     *
+     * @param \Magento\NewsletterApi\Api\Data\SubscriptionInterface $subscription the subscription
+     *
+     * @return bool true on success
+     *
+     * @throws \Magento\NewsletterApi\Exception\AlreadySubscribedException when subscription already exists
+     * @throws \Magento\NewsletterApi\Exception\CouldNotSubscribeException if an error occurred during subscription
+     * @throws \Magento\Framework\Exception\InputException when an invalid input has been provided
+     */
+    public function subscribe(Data\SubscriptionInterface $subscription);
+
+    /**
+     * Subscribe a customer
+     *
+     * Adds a new Newsletter Subscription for the given customer
+     *
+     * @param \Magento\Customer\Api\Data\CustomerInterface $customer the customer to be subscribed
+     *
+     * @return bool true on success
+     *
+     * @throws \Magento\NewsletterApi\Exception\AlreadySubscribedException when subscription already exists
+     * @throws \Magento\NewsletterApi\Exception\CouldNotSubscribeException if an error occurred during subscription
+     * @throws \Magento\Framework\Exception\InputException when an invalid input has been provided
+     */
+    public function subscribeCustomer(CustomerInterface $customer);
+
+    /**
+     * Subscribe a customer by id
+     *
+     * Adds a new Newsletter Subscription for the given customer
+     * by provided customer id
+     *
+     * @param int $customerId the id of the customer to be subscribed
+     *
+     * @return bool true on success
+     *
+     * @throws \Magento\NewsletterApi\Exception\AlreadySubscribedException when subscription already exists
```

Comment:
Since there is no easy way to check if customer/email is already subscribed, throwing the exception if user is already subscribed will promote using exceptions for conditional logic. Do not throw the exception.